src/Elastic.Transport/Products/Elasticsearch/ElasticsearchProductRegistration.cs

```diff
@@ -20,7 +21,7 @@ public class ElasticsearchProductRegistration : ProductRegistration
 {
 	private readonly HeadersList _headers;
 	private readonly MetaHeaderProvider _metaHeaderProvider;
-	private readonly int? _clientMajorVersion;
+	private readonly BigInteger? _clientMajorVersion;
```

Comment:
Is dit mandated by the new server version? Surprised this is needed :)

Reply:
Also surprised by this. Could we continue to `int` and cast from the SemVer property? Our versions are never going to get that large!!

Reply:
This is not really needed at the moment (probably never 😄 ), but the `SemVersion` type uses `BigInt` for the `Major`, `Minor` and `Patch` types so it makes sense for me to use the same type here instead of converting it to `int` as this field is `private` anyways.

Reply:
Refactored the code to use `int` even internally.